node/node_test.go

```diff
@@ -0,0 +1,83 @@
+package node
+
+import (
+	"context"
+	"fmt"
+	"testing"
+
+	proxy "github.com/cometbft/cometbft/proxy"
+	cmtypes "github.com/cometbft/cometbft/types"
+	"github.com/stretchr/testify/assert"
+	"github.com/stretchr/testify/require"
+
+	"github.com/rollkit/rollkit/config"
+	test "github.com/rollkit/rollkit/test/log"
+	"github.com/rollkit/rollkit/types"
+)
+
+// cleanUpNode stops the node and checks if it is running
+func cleanUpNode(node Node, t *testing.T) {
+	defer func() {
+		if fn, ok := node.(*FullNode); ok {
+			fn.cancel()
+		}
+		if ln, ok := node.(*LightNode); ok {
+			ln.cancel()
+		}
+	}()
+	assert.NoError(t, node.Stop())
+	assert.False(t, node.IsRunning())
+}
+
+func initializeAndStartFullNode(ctx context.Context, t *testing.T) *FullNode {
+	node := initializeAndStartNode(ctx, t, "full")
+	return node.(*FullNode)
+}
+func initializeAndStartLightNode(ctx context.Context, t *testing.T) *LightNode {
+	node := initializeAndStartNode(ctx, t, "light")
+	return node.(*LightNode)
+}
+
+// initializeAndStartNode initializes and starts a test node
+func initializeAndStartNode(ctx context.Context, t *testing.T, nodeType string) Node {
+	node := setupTestNode(ctx, t, nodeType)
+	require.False(t, node.IsRunning())
+	err := node.Start()
+	require.NoError(t, err)
+	require.True(t, node.IsRunning())
+	return node
+}
+
+func newTestNode(ctx context.Context, t *testing.T, nodeType string) (Node, error) {
+	config := config.NodeConfig{DALayer: "mock"}
+	switch nodeType {
+	case "light":
+		config.Light = true
+	case "full":
+		config.Light = false
+	default:
+		panic(fmt.Sprint("invalid node type", nodeType))
```

Comment:
Instead of panicking, shall we use `t.Error` instead since that's more canonical?

Reply:
The reason I use panics here is because it is a developer error. I think it makes it more explicit that the developer needs to update how they wrote the test, vs seeing a test failure. 

This could be a low priority follow up to discuss with the team how we want to handle these types of situations and ensure the rest of the test suite is aligned.

Reply:
Follow up issue: https://github.com/rollkit/rollkit/issues/1321